sbol_utilities/helper_functions.py

```diff
@@ -37,26 +38,52 @@ def id_sort(i: iter):
     return sorted(i, key=lambda x: x.identity if isinstance(x, sbol3.Identified) else x)
 
 
-def find_child(ref: ReferencedURI):
+def build_reference_cache(doc: sbol3.Document) -> dict[str, sbol3.Identified]:
+    """Build a cache of identities from the given document to support
+    faster lookups of referenced objects.
+
+    :param doc: an sbol3 Document
+    :returns: a cache of identities
+    """
+    cache = {}
+    def cache_identity(object: sbol3.Identified):
+        cache[object.identity] = object
+    doc.traverse(cache_identity)
+    return cache
+
+
+def find_child(ref: ReferencedURI, cache: Optional[dict[str, sbol3.Identified]] = None):
     """Look up a child object; if it is not found, raise an exception
 
     :param ref: reference to look up
     :returns: object pointed to by reference
     :raises ChildNotFound: if object cannot be retrieved
     """
+    try:
+        return cache[str(ref)]
+    except KeyError:
```

Comment:
These exceptions could use a comment explaining why we're passing on them. Same below in find_top_level.

Reply:
@jakebeal This is a great point. I added comments explaining what the raised exceptions probably mean, and that ignoring them falls through to the expensive lookup.